examples/source_separation/lightning_trainer.py

```diff
@@ -0,0 +1,432 @@
+#!/usr/bin/env python3
+
+# pyre-strict
+
+import pathlib
+from argparse import ArgumentParser
+from typing import (
+    Any,
+    Dict,
+    Mapping,
+    List,
+    Optional,
+    Tuple,
+    TypedDict,
+)
+
+import torch
+import torchaudio
+import torchaudio.models
+from pytorch_lightning import LightningModule, Trainer
+from pytorch_lightning.callbacks import ModelCheckpoint, EarlyStopping
+from pytorch_lightning.plugins import DDPPlugin
+from torch import nn
+from torch.optim.lr_scheduler import _LRScheduler
```

Comment:
nit for the PyTorch team: this makes me think that we should probably expose a public class for the schedulers, it's generally not good practice to rely on private functions.

cc @datumbox @gchanan for visibility